docs/cudf/source/developer_guide/pylibcudf.md

```diff
@@ -0,0 +1,113 @@
+
+For example, `libcudf::data_type` maps to `pylibcudf.DataType`, which looks like this (implementation omitted):
+```cython
+
+cdef class DataType:
+    cdef data_type c_obj
+
+    cpdef TypeId id(self)
+    cpdef int32_t scale(self)
+
+    @staticmethod
+    cdef DataType from_data_type(data_type dt)
+```
+
+This allows pylibcudf functions to accept a typed `DataType` parameter and then trivially call underlying libcudf algorithms by accessing the argument's `c_obj`.
+
+#### pylibcudf Tables and Columns
+
+The primary exception to the above set of rules are libcudf's core data owning types, `cudf::table` and `cudf::column`.
+libcudf uses modern C++ idioms based on smart pointers to avoid resource leaks and make code exception-safe.
+To avoid passing around raw pointers, and to ensure that ownership semantics are clear, libcudf has separate `view` types corresponding to data owning types.
+For example, `cudf::column` owns data, while `cudf::column_view` represents an view on a column of data, while `cudf::mutable_column_view` represents a mutable view.
+A `column_view` need not actually reference data owned by a `cudf::column`; any memory buffer will do.
+This separate allows libcudf algorithms to clearly communicate ownership expectations and allows multiple views into the same data to coexist.
+
+While libcudf algorithms accept views as inputs, any algorithms that allocate data must return `cudf::column` and `cudf::table` objects.
+libcudf's ownership model is problematic for pylibcudf, which must be able to seamlessly interoperate with data provided by other Python libraries like PyTorch or Numba.
+Therefore, pylibcudf employs the following strategy:
+- pylibcudf defines the `gpumemoryview` type, which (analogous to the [Python `memoryview` type](https://docs.python.org/3/library/stdtypes.html#memoryview)) represents a view into memory owned by another object that it keeps alive using Python's standard reference counting machinery. A `gpumemoryview` is constructible from any object implementing the [CUDA Array Interface protocol](https://numba.readthedocs.io/en/stable/cuda/cuda_array_interface.html).
+  - This type will eventually be generalized for reuse outside of pylibcudf.
+- pylibcudf defines its own Table and Column classes.
+  - A Table maintains Python references to the Columns it contains, so multiple Tables may share the same Column.
+  - A Column consists of `gpumemoryview`s of its data buffers (which may include children for nested types) and its null mask.
+- `pylibcudf.Table` and `pylibcudf.Column` provide easy access to `cudf::table_view` and `cudf::column_view` objects viewing the same columns/memory. These can be then be used when implementing any pylibcudf algorithm in terms of the underlying libcudf algorithm. Specifically, each of these classes owns an instance of the libcudf view type and provides a method `get_underlying` that may be used to access a pointer to that object to be passed to libcudf.
```

Comment:
Should this instead (going with the mirroring) be called `view` (and presumably, `mutable_view`)?

Reply:
Yup, changed it.